dallinger/bots.py

```diff
@@ -226,15 +227,12 @@ def sign_up(self):
                 )
             )
             result = requests.post(url)
-            if result.status_code == 500:
-                self.stochastic_sleep()
-                continue
-            elif result.json()['status'] == 'error':
+            if result.status_code == 500 or result.json()['status'] == 'error':
                 self.stochastic_sleep()
                 continue
-            else:
-                self.participant_id = result.json()['participant']['id']
-                return self.participant_id
+
+            self.upon_signup(result.json())
```

Comment:
Hmm, `upon_signup` here but `on_signup` below?

Reply:
Fixed. Still getting the hang of VS Code, apparently.